recipes/mosquitto/2.x/conanfile.py

```diff
@@ -155,26 +158,26 @@ def package_info(self):
             elif self.settings.os == "Windows":
                 self.cpp_info.components["libmosquittopp"].system_libs = ["ws2_32"]
 
-        if self.options.broker:
-            self.cpp_info.components["broker"].libdirs = []
-            self.cpp_info.components["broker"].include_dirs = []
+        if self.options.broker or self.options.get_safe("apps") or self.options.get_safe("clients"):
             bin_path = os.path.join(self.package_folder, "bin")
             self.output.info("Appending PATH env var with : {}".format(bin_path))
             self.env_info.PATH.append(bin_path)
+
+        if self.options.broker:
+            self.cpp_info.components["mosquitto_broker"].libdirs = []
+            self.cpp_info.components["mosquitto_broker"].include_dirs = []
             if self.options.websockets:
-                self.cpp_info.components["broker"].requires.append("libwebsockets::libwebsockets")
+                self.cpp_info.components["mosquitto_broker"].requires.append("libwebsockets::libwebsockets")
             if self.settings.os in ("FreeBSD", "Linux"):
-                self.cpp_info.components["broker"].system_libs = ["pthread", "m"]
+                self.cpp_info.components["mosquitto_broker"].system_libs = ["pthread", "m"]
             elif self.settings.os == "Windows":
-                self.cpp_info.components["broker"].system_libs = ["ws2_32"]
+                self.cpp_info.components["mosquitto_broker"].system_libs = ["ws2_32"]
 
         for option in ["apps", "clients"]:
             if self.options.get_safe(option):
-                self.cpp_info.components[option].libdirs = []
-                self.cpp_info.components[option].include_dirs = []
-                bin_path = os.path.join(self.package_folder, "bin")
-                self.output.info("Appending PATH env var with : {}".format(bin_path))
-                self.env_info.PATH.append(bin_path)
-                self.cpp_info.components[option].requires = ["openssl::openssl", "libmosquitto"]
+                option_comp_name = "mosquitto_{}".format(option)
+                self.cpp_info.components[option_comp_name].libdirs = []
+                self.cpp_info.components[option_comp_name].include_dirs = []
+                self.cpp_info.components[option_comp_name].requires = ["openssl::openssl", "libmosquitto"]
                 if self.options.cjson:
-                    self.cpp_info.components[option].requires.append("cjson::cjson")
+                    self.cpp_info.components[option_comp_name].requires.append("cjson::cjson")
```

Comment:
What if this is built with: `-o mosquitto:apps=False -o mosquitto:clients=False -o mosquitto:cjson=True`?

Reply:
options.cjson is deleted in configure() if clients if False, avoiding to depend on cjson in requirements()

Reply:
So you can ask: what happen when clients if False, apps is True?

And in the lines above: what happend when ssl option is False?

Reply:
Time to introduce a `_with_cjson`/`_with_ssl` method?
Assuming cmake supports building the library with no apps/clients and no cjson present.

Reply:
yes, to keep in mind for another PR.